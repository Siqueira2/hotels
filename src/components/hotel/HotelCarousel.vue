<template>
  <q-carousel
      v-model="slide"
      animated
      arrows
      navigation
      infinite
      :autoplay="10000"
      :style="getHeightStyle"
    >
      <q-carousel-slide
        v-for="(image, index) in hotel.getImages"
        :key="index"
        :name="index"
        :style="getHeightStyle"
        class="q-pa-none overflow-hidden"
      >
        <q-img :src="image" basic class="full-width" :style="getHeightStyle" fit="cover" />
    </q-carousel-slide>
  </q-carousel>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue'

import type { Hotel } from 'src/models'

const props = defineProps<{
  hotel: Hotel
  height?: number
}>()

const slide = ref(0)

const getHeightStyle = computed(() => {
  const carouselHeight = props.height || 256

  return `height: ${carouselHeight}px`
})
</script>
