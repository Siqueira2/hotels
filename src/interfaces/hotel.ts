import type { Amenity } from './amenity'

export interface PlaceHotels {
  placeId: number
  hotels: Hotel[]
}

export interface Address {
  fullAddress: string
  city: string
  district: string
  street: string
  state: string
  number: string
  country: string
  zipCode: string | null
}

export interface Hotel {
  id: number
  favorite: boolean
  name: string
  description: string
  stars: string
  thumb: string
  amenities: Amenity[] | null
  hasBreakFast: boolean
  hasRefundableRoom: boolean
  hasAgreement: boolean
  nonRefundable: boolean | null
  address: Address
  images: string[]
  deals: unknown
  roomsQuantity: number
  price: number
}

export interface FetchHotelsParams {
  page?: number
  limit?: number
  name?: string
  sortBy?: string
}
