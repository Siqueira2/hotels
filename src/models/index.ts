export * from './place'
export * from './hotel'
export * from './placeHotel'
export * from './hotelFilterService'
