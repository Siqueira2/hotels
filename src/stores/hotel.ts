import { defineStore } from 'pinia'
import { ref } from 'vue'

import { fetchPlaceHotels } from 'src/api/hotels'

import type { Hotel as HotelModel } from 'src/models/hotel'

export const useHotelStore = defineStore('hotels', () => {
  const hotels = ref<HotelModel[]>([])

  const fetchHotels = async () => {
    const response = await fetchPlaceHotels()
    hotels.value = response.flatMap(({ hotels }) => hotels)
  }

  const getRecommendedHotels = () => {
    return hotels.value
      .filter((hotel) => hotel.getHasBreakfast && hotel.hasAmenities)
      .sort((a, b) => b.getPrice - a.getPrice)
      .slice(0, 5)
  }

  const getBestRatedHotels = () => {
    return hotels.value.sort((a, b) => Number(b.getStars) - Number(a.getStars)).slice(0, 5)
  }

  return {
    hotels,
    fetchHotels,
    getRecommendedHotels,
    getBestRatedHotels,
  }
})
