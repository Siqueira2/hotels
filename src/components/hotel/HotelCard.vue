<template>
  <q-card flat class="q-mt-md row">
    <q-carousel
      class="col-12 col-sm-3"
      v-model="slide"
      animated
      arrows
      navigation
      infinite
      :autoplay="10000"
      style="height: 256px"
    >
      <q-carousel-slide
        v-for="(image, index) in hotel.getImages"
        :key="index"
        :name="index"
        style="height: 256px"
        class="q-pa-none overflow-hidden"
      >
        <q-img :src="image" basic class="full-width" style="height: 256px" fit="cover" />
      </q-carousel-slide>
    </q-carousel>

    <q-card-section class="col-12 col-sm-5 col-md-6">
      <h6 class="text-grey-2 q-ma-none text-weight-regular">
        {{ hotel.getName }}
      </h6>

      <p class="text-subtitle2 text-grey-5 q-ma-none text-weight-regular">
        {{ hotel.getLocation }}
      </p>

      <div class="row">
        <div class="col-12 q-mt-sm">
          <div class="row items-center">
            <span class="text-grey-6 text-caption">{{ hotel.getStars }}</span>
            <span class="text-yellow-8 q-ml-xs">{{ hotel.getStarRating }}</span>

            <q-separator vertical color="grey-6" class="q-mx-sm" />

            <template v-if="hotel.hasAmenities">
              <q-icon
                v-for="amenity in hotel.getAmenities"
                :key="amenity.key"
                :name="getAmenityIcon(amenity.key)"
                size="xs"
                color="grey-6"
                class="q-mr-sm"
              >
                <q-tooltip>{{ amenity.label }}</q-tooltip>
              </q-icon>
            </template>
          </div>
        </div>
      </div>

      <q-chip
        v-if="hotel.isRefundable"
        color="grey-6"
        text-color="white"
        size="sm"
        square
        class="q-mx-none q-mt-sm"
      >
        {{ t('hotels.hotel.refundable') }}
      </q-chip>
    </q-card-section>

    <q-card-section class="col-12 col-sm-4 col-md-3">
      <div class="text-grey-5 text-weight-regular">
        <p class="q-ma-none text-weight-regular">
          {{ t('hotels.hotel.from') }}
        </p>

        <h5 class="text-grey-2 q-ma-none">
          <span class="text-caption">{{ t('shared.currency') }}</span>

          <span class="text-weight-semibold">{{ hotel.getPrice.toFixed(2) }}</span>
        </h5>

        <p class="text-grey-6 text-caption q-ma-none">
          <span>{{ $t('shared.currency') }}</span>

          {{ getPricePerNightText }}
        </p>

        <p class="text-grey-6 text-subtitle2 text-weight-semibold">
          {{ t('hotels.hotel.included-taxes') }}
        </p>
      </div>

      <q-btn
        color="primary"
        :label="$t('shared.actions.select')"
        rounded
        unelevated
        class="q-mt-lg"
        @click="$emit('select', hotel)"
      />
    </q-card-section>
  </q-card>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue'
import { useI18n } from 'vue-i18n'

import type { Hotel } from 'src/models/hotel'
import { AmenitiesEnum } from 'src/enums'

const { hotel } = defineProps<{
  hotel: Hotel
}>()

const { t } = useI18n()

const slide = ref(0)

const getPricePerNightText = computed(() =>
  t('hotels.hotel.price-per-night', { price: hotel.getPricePerNight }),
)

const getAmenityIcon = (amenity: keyof typeof AmenitiesEnum) => {
  return AmenitiesEnum[amenity] || 'mdi-help'
}

defineEmits<{
  select: [hotel: Hotel]
}>()
</script>

<style scoped lang="scss">
.q-carousel {
  border-top-right-radius: 0 !important;
  border-bottom-right-radius: 0 !important;
  border-bottom-left-radius: inherit !important;
}

.q-card__section:last-child {
  border-left: 2px solid $grey-1 !important;
  border-bottom-left-radius: 0 !important;
}
</style>
