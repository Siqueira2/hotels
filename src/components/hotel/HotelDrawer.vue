<template>
  <div
    v-if="isDrawerOpen"
    @click="isDrawerOpen = false"
    class="absolute-top-left full-width full-height backdrop"
  />

  <q-drawer overlay v-model="isDrawerOpen" side="right" bordered :width="800">
    <q-card flat>
      <q-card-section>
        <div>
          <span class="text-grey-6 text-caption">{{ hotel.getStars }}</span>

          <span class="text-yellow-8 q-ml-xs">{{ hotel.getStarRating }}</span>
        </div>

        <h5 class="text-grey-2 q-ma-none text-weight-regular">
          {{ hotel.getName }}
        </h5>

        <p class="text-subtitle2 text-grey-5 q-ma-none text-weight-regular">
          {{ hotel.getLocation }}
        </p>
      </q-card-section>

      <q-carousel
        v-model="slide"
        animated
        arrows
        navigation
        infinite
        :autoplay="10000"
        class="q-mx-md"
        style="height: 400px"
      >
        <q-carousel-slide
          v-for="(image, index) in hotel.getImages"
          :key="index"
          :name="index"
          style="height: 400px"
          class="q-pa-none overflow-hidden"
        >
          <q-img :src="image" basic class="full-width" style="height: 400px" fit="cover" />
        </q-carousel-slide>
      </q-carousel>

      <q-card-section v-if="hotel.hasAmenities">
        <h6 class="text-grey-2 q-mt-none q-mb-md text-weight-regular">
          {{ t('hotels.hotel.drawer.amenities-title') }}
        </h6>
        <q-separator />

        <div class="row q-gutter-x-xl q-my-md" data-testid="hotel-amenities">
          <div v-for="amenity in getHotelAmenities" :key="amenity.key" class="row q-mb-sm">
            <q-icon :name="getAmenityIcon(amenity.key)" size="xs" color="grey-6" class="q-mr-sm" />

            <p class="text-grey-6 q-mb-none">{{ amenity.label }}</p>
          </div>
        </div>

        <div v-if="showAmenitiesButton" class="w-full flex justify-center q-mb-md">
          <q-btn
            color="primary"
            :label="getAmenitiesButtonLabel"
            outline
            rounded
            @click="showAmenities = !showAmenities"
          />
        </div>

        <q-separator />
      </q-card-section>

      <q-card-section class="q-py-none">
        <h6 class="text-grey-2 q-mt-none q-mb-md text-weight-regular">
          {{ t('hotels.hotel.drawer.description') }}
        </h6>

        <p class="text-grey-5" data-testid="hotel-description">
          {{ hotel.getDescription }}
        </p>
      </q-card-section>
    </q-card>
  </q-drawer>
</template>

<script setup lang="ts">
import { computed, ref } from 'vue'
import { useI18n } from 'vue-i18n'

import type { Hotel } from 'src/models/hotel'

import { AmenitiesEnum } from 'src/enums'

const { t } = useI18n()

const { hotel, modelValue } = defineProps<{
  hotel: Hotel
  modelValue: boolean
}>()

const emit = defineEmits<{
  'update:modelValue': [value: boolean]
}>()

const slide = ref(0)
const showAmenities = ref(false)
const amenitiesLimit = ref(4)

const getHotelAmenities = computed(() => {
  return showAmenities.value
    ? hotel.getAmenities
    : hotel.getAmenities?.slice(0, amenitiesLimit.value)
})

const showAmenitiesButton = computed(() => {
  if (hotel.getAmenities?.length) {
    return hotel.getAmenities?.length > amenitiesLimit.value
  }

  return false
})

const getAmenitiesButtonLabel = computed(() => {
  return showAmenities.value
    ? t('hotels.hotel.drawer.show-less')
    : t('hotels.hotel.drawer.show-all')
})

const isDrawerOpen = computed({
  get: () => modelValue,
  set: (value) => emit('update:modelValue', value),
})

const getAmenityIcon = (amenity: keyof typeof AmenitiesEnum) => {
  return AmenitiesEnum[amenity] || 'mdi-help'
}
</script>

<style scoped lang="scss">
.backdrop {
  background-color: $dark;
  opacity: 0.7;
}

.q-carousel {
  border-radius: 16px !important;
}
</style>
