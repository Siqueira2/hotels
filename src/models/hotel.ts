import type { Hotel as HotelType, Address, Amenity } from 'src/interfaces'

import { formatCurrency } from 'src/helpers'

export class Hotel {
  private id: number
  private name: string
  private description: string
  private stars: string
  private thumb: string
  private favorite: boolean
  private price: number
  private address: Address
  private amenities: Amenity[] | null
  private hasBreakFast: boolean
  private hasRefundableRoom: boolean
  private hasAgreement: boolean
  private nonRefundable: boolean | null
  private roomsQuantity: number
  private images: string[]

  constructor(hotel: HotelType) {
    this.id = hotel.id
    this.name = hotel.name
    this.description = hotel.description
    this.stars = hotel.stars
    this.thumb = hotel.thumb
    this.favorite = hotel.favorite
    this.price = hotel.price
    this.address = hotel.address
    this.amenities = hotel.amenities
    this.hasBreakFast = hotel.hasBreakFast
    this.hasRefundableRoom = hotel.hasRefundableRoom
    this.hasAgreement = hotel.hasAgreement
    this.nonRefundable = hotel.nonRefundable
    this.roomsQuantity = hotel.roomsQuantity
    this.images = hotel.images
  }

  public get getId(): number {
    return this.id
  }

  public get getName(): string {
    return this.name
  }

  public get getDescription(): string {
    return this.description
  }

  public get getStarRating(): string {
    return 'â˜…'.repeat(Number(this.stars))
  }

  public get getLocation(): string {
    return `${this.address.fullAddress}, ${this.address.city}, ${this.address.state}, ${this.address.zipCode || 'N/A'}`
  }

  public get isFavorite(): boolean {
    return this.favorite
  }

  public get hasAmenities(): boolean {
    return this.amenities !== null && this.amenities.length > 0
  }

  public get getPrice(): string {
    return formatCurrency(this.price)
  }

  public get getHasBreakfast(): boolean {
    return this.hasBreakFast
  }

  public get getStars(): string {
    return this.stars
  }

  public get getThumb(): string {
    return this.thumb
  }

  public get getImages(): string[] {
    return this.images
  }

  public get isRefundable(): boolean {
    return this.hasRefundableRoom && !this.nonRefundable
  }

  public get isAgreement(): boolean {
    return this.hasAgreement
  }

  public get getRoomsQuantity(): number {
    return this.roomsQuantity
  }

  public get getAmenities(): Amenity[] | null {
    return this.amenities
  }

  public get getPricePerNight(): string {
    return formatCurrency(Number(this.price / this.roomsQuantity))
  }
}
