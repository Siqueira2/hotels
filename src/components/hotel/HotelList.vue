<template>
  <q-infinite-scroll @load="onLoad">
    <div v-if="loading" class="row justify-center q-my-md">
      <q-spinner-dots color="primary" size="40px" />
    </div>

    <template v-else-if="hotels.length">
      <HotelCard
        v-for="hotel in hotels"
        :key="hotel.getId"
        :hotel="hotel as Hotel"
        @select="openDrawer"
      />
    </template>

    <div v-else class="row justify-center q-my-md">
      <p class="text-grey-2">{{ $t('hotels.no-hotels') }}</p>
    </div>

    <template #loading>
      <div v-if="loading || loadingNextPage" class="row justify-center q-my-md">
        <q-spinner-dots color="primary" size="40px" />
      </div>
    </template>
  </q-infinite-scroll>

  <HotelDrawer v-if="selectedHotel" :hotel="selectedHotel as Hotel" v-model="drawerOpen" />
</template>

<script setup lang="ts">
import { ref, onMounted, watch } from 'vue'
import { storeToRefs } from 'pinia'
import { useRoute } from 'vue-router'

import { useHotelStore } from 'src/stores/hotel'
import type { Hotel } from 'src/models'

import HotelCard from 'src/components/hotel/HotelCard.vue'
import HotelDrawer from 'src/components/hotel/HotelDrawer.vue'

const route = useRoute()
const hotelStore = useHotelStore()
const { hotels } = storeToRefs(hotelStore)

const page = ref(1)
const loading = ref(true)
const loadingNextPage = ref(false)
const drawerOpen = ref(false)
const selectedHotel = ref<Hotel | null>(null)

const fetchHotels = async (params: { page: number }) => {
  await hotelStore.fetchHotels({
    page: params.page,
    name: route.query.name as string,
    sortBy: route.query.sort as string,
  })
}

const onLoad = async (_: number, done: VoidFunction) => {
  if (!loading.value) {
    loadingNextPage.value = true
    page.value++
    await fetchHotels({ page: page.value })
    loadingNextPage.value = false
  }
  done()
}

const openDrawer = (hotel: Hotel) => {
  selectedHotel.value = hotel
  drawerOpen.value = true
}

onMounted(async () => {
  loading.value = true
  await fetchHotels({ page: page.value })
  loading.value = false
})

watch(
  () => route.query,
  async () => {
    page.value = 1
    loading.value = true
    await fetchHotels({ page: 1 })
    loading.value = false
  },

  { deep: true },
)
</script>
