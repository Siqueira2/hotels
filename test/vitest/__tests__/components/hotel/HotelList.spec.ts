import { describe, it, expect, vi, beforeEach } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'

import type { Hotel as HotelType } from 'src/interfaces'

import { useHotelStore } from 'src/stores/hotel'

import HotelList from 'src/components/hotel/HotelList.vue'
import HotelCard from 'src/components/hotel/HotelCard.vue'

import HOTEL from 'src/data/hotel.json'
import { Hotel } from 'src/models/hotel'

vi.mock('vue-router', () => {
  return {
    useRoute: () => ({ query: {} }),
    useRouter: () => ({ push: vi.fn() }),
  }
})

const mockHotel = new Hotel(HOTEL[0]?.hotels[0] as HotelType)
const createWrapper = () =>
  mount(HotelList, {
    global: {
      plugins: [createTestingPinia({ createSpy: vi.fn })],
    },
  })

describe('HotelList', () => {
  let wrapper: VueWrapper
  let store: ReturnType<typeof useHotelStore>

  beforeEach(() => {
    wrapper = createWrapper()
    store = useHotelStore()
    store.hotels = [mockHotel]
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should call fetchHotels on mount', () => {
    expect(store.fetchHotels).toHaveBeenCalled()
  })

  it('should render hotel list', () => {
    expect(wrapper.findAllComponents(HotelCard)).toHaveLength(1)
  })

  it('should render hotel card', () => {
    expect(wrapper.findComponent(HotelCard).exists()).toBeTruthy()
  })

  it('should call fetchHotel infinite scroll', async () => {
    const infiniteScroll = wrapper.find('.q-infinite-scroll')
    await infiniteScroll.trigger('load')

    expect(store.fetchHotels).toHaveBeenCalled()
  })
})
