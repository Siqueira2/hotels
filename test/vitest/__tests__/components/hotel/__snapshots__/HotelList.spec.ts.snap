// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HotelList > should match snapshot 1`] = `
"<div class=\\"q-infinite-scroll\\">
  <div data-v-be666f43=\\"\\" class=\\"q-card q-card--flat no-shadow q-mt-md row\\">
    <div data-v-be666f43=\\"\\" class=\\"q-carousel q-panel-parent q-carousel--without-padding q-carousel--arrows-horizontal q-carousel--navigation-bottom col-12 col-sm-3 carousel\\" style=\\"height: 256px;\\">
      <div class=\\"q-carousel__slides-container\\">
        <transition-stub appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"q-panel scroll\\" style=\\"--q-transition-duration: 300ms;\\" role=\\"tabpanel\\">
            <div class=\\"q-carousel__slide q-pa-none overflow-hidden\\" style=\\"height: 256px;\\">
              <div class=\\"q-img q-img--menu full-width\\" style=\\"height: 256px;\\" role=\\"img\\" basic=\\"\\">
                <div style=\\"padding-bottom: 56.24929688378895%;\\"></div>
                <div class=\\"q-img__container absolute-full\\"><img class=\\"q-img__image q-img__image--with-transition q-img__image--current\\" style=\\"object-fit: cover; object-position: 50% 50%;\\" fetchpriority=\\"auto\\" aria-hidden=\\"true\\" draggable=\\"false\\" src=\\"https://s3.amazonaws.com/e-htl/uploads/hotels/10415/10415_10.JPG\\"></div>
                <transition-stub name=\\"q-transition--fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <div class=\\"q-img__loading absolute-full flex flex-center\\"><svg class=\\"q-spinner q-spinner-mat\\" width=\\"1em\\" height=\\"1em\\" viewBox=\\"25 25 50 50\\">
                      <circle class=\\"path\\" cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"5\\" stroke-miterlimit=\\"10\\"></circle>
                    </svg></div>
                </transition-stub>
              </div>
            </div>
          </div>
        </transition-stub>
      </div>
      <div class=\\"q-carousel__control q-carousel__navigation no-wrap absolute flex q-carousel__navigation--buttons q-carousel__navigation--bottom\\">
        <div class=\\"q-carousel__navigation-inner flex flex-center no-wrap\\">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div class=\\"q-carousel__control q-carousel__arrow q-carousel__prev-arrow q-carousel__prev-arrow--horizontal absolute flex flex-center\\"><button class=\\"q-btn q-btn-item non-selectable no-outline q-btn--flat q-btn--round q-btn--actionable q-focusable q-hoverable q-btn--dense\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\"><i class=\\"q-icon notranslate material-icons\\" aria-hidden=\\"true\\" role=\\"img\\">chevron_left</i></span></button></div>
      <div class=\\"q-carousel__control q-carousel__arrow q-carousel__next-arrow q-carousel__next-arrow--horizontal absolute flex flex-center\\"><button class=\\"q-btn q-btn-item non-selectable no-outline q-btn--flat q-btn--round q-btn--actionable q-focusable q-hoverable q-btn--dense\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\"><i class=\\"q-icon notranslate material-icons\\" aria-hidden=\\"true\\" role=\\"img\\">chevron_right</i></span></button></div>
    </div>
    <div data-v-be666f43=\\"\\" class=\\"q-card__section q-card__section--vert col-12 col-sm-5 col-md-6\\">
      <h6 data-v-be666f43=\\"\\" class=\\"text-grey-2 q-ma-none text-weight-regular\\">Nacional Inn Belo Horizonte</h6>
      <p data-v-be666f43=\\"\\" class=\\"text-subtitle2 text-grey-5 q-ma-none text-weight-regular\\">Rua Espírito Santo, 215 - Centro, Belo Horizonte, MG, N/A</p>
      <div data-v-be666f43=\\"\\" class=\\"row\\">
        <div data-v-be666f43=\\"\\" class=\\"col-12 q-mt-sm\\">
          <div data-v-be666f43=\\"\\" class=\\"row items-center\\"><span data-v-be666f43=\\"\\" class=\\"text-grey-6 text-caption\\">3</span><span data-v-be666f43=\\"\\" class=\\"text-yellow-8 q-ml-xs\\">★★★</span>
            <hr data-v-be666f43=\\"\\" class=\\"q-separator q-separator--vertical bg-grey-6 q-mx-sm\\" aria-orientation=\\"vertical\\"><i data-v-be666f43=\\"\\" class=\\"q-icon text-grey-6 notranslate material-icons q-mr-sm\\" style=\\"font-size: 18px;\\" aria-hidden=\\"true\\" role=\\"presentation\\">wifi
              <!---->
            </i><i data-v-be666f43=\\"\\" class=\\"q-icon text-grey-6 notranslate material-icons q-mr-sm\\" style=\\"font-size: 18px;\\" aria-hidden=\\"true\\" role=\\"presentation\\">restaurant
              <!---->
            </i><i data-v-be666f43=\\"\\" class=\\"q-icon text-grey-6 notranslate material-icons q-mr-sm\\" style=\\"font-size: 18px;\\" aria-hidden=\\"true\\" role=\\"presentation\\">room_service
              <!---->
            </i>
          </div>
        </div>
      </div>
      <div data-v-be666f43=\\"\\" class=\\"q-chip row inline no-wrap items-center bg-grey-6 text-white q-chip--colored q-chip--square q-mx-none q-mt-sm\\" style=\\"font-size: 10px;\\">
        <div class=\\"q-chip__content col row no-wrap items-center q-anchor--skip\\">Reembolsável</div>
      </div>
    </div>
    <div data-v-be666f43=\\"\\" class=\\"q-card__section q-card__section--vert col-12 col-sm-4 col-md-3\\">
      <div data-v-be666f43=\\"\\" class=\\"text-grey-5 text-weight-regular\\">
        <p data-v-be666f43=\\"\\" class=\\"q-ma-none text-weight-regular\\">A partir de</p>
        <h5 data-v-be666f43=\\"\\" class=\\"text-grey-2 q-ma-none\\"><span data-v-be666f43=\\"\\" class=\\"text-caption q-mr-xs\\">shared.currency</span><span data-v-be666f43=\\"\\" class=\\"text-weight-semibold\\">100,00</span></h5>
        <p data-v-be666f43=\\"\\" class=\\"text-grey-6 text-caption q-ma-none\\">R$ 50,00/noite</p>
        <p data-v-be666f43=\\"\\" class=\\"text-grey-6 text-subtitle2 text-weight-semibold\\">Imposto incluídos</p>
      </div><button data-v-be666f43=\\"\\" class=\\"q-btn q-btn-item non-selectable no-outline q-btn--unelevated q-btn--rectangle q-btn--rounded bg-primary text-white q-btn--actionable q-focusable q-hoverable q-mt-lg\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\"><span class=\\"block\\">shared.actions.select</span></span></button>
    </div>
  </div>
  <div class=\\"q-infinite-scroll__loading invisible\\">
    <!--v-if-->
  </div>
</div>
<!--v-if-->"
`;
